lib/std/crypto/ripemd160.zig

```diff
@@ -0,0 +1,218 @@
+// https://homes.esat.kuleuven.be/~bosselae/ripemd160.html
+
+const std = @import("std");
+
+pub const Ripemd160 = struct {
+    bytes: [160 / 8]u8,
+
+    pub fn hash(str: []const u8) Ripemd160 {
+        var res: Ripemd160 = undefined;
+        var temp: [5]u32 = .{ 0x67452301, 0xefcdab89, 0x98badcfe, 0x10325476, 0xc3d2e1f0 };
+        var ptr = str.ptr;
+
+        { // compress string in 16-word chunks
+            var i = str.len / 64;
+            while (i > 0) : (i -= 1) {
+                var chunk: [16]u32 = [_]u32{0} ** 16;
+                var j: usize = 0;
+                while (j < 16) : (j += 1) {
+                    std.mem.copy(u8, @ptrCast([*]u8, &chunk[j])[0..4], ptr[0..4]);
+                    ptr += 4;
```

Comment:
That looks too much like a straight conversion from C code. You can simply use an index, and `mem.readIntLittle(u32, ...)` to read a 32-bit integer at a given offset.